hbase-client/src/main/java/org/apache/hadoop/hbase/filter/PrefixFilter.java

```diff
@@ -76,8 +76,9 @@ public boolean filterRowKey(Cell firstRowCell) {
     if (filterAllRemaining()) {
       return true;
     }
+    // if the cell is smaller => return false so that getNextCellHint() is invoked.
```

Comment:
nit:
For reverse scans, smaller/bigger is not accurate.
Maybe use before/after instead.